jazz-build-module/environment-deployment-metadata-loader.groovy

```diff
@@ -0,0 +1,499 @@
+					}
+				}
+			}
+		}
+        return env_logical_ids
+	}
+	catch(e){
+		echo "error occured while fetching environment id list: " + e.getMessage()
+		return env_logical_ids
+	}
+}
+
+def getEnvironmentBranchName(logical_id) {
+	def branchName
+	try {
+		def environment_data = sh (script: "curl GET  \
+			-H \"Content-Type: application/json\" \
+			-H \"Authorization: $g_login_token\" \
+			\"${g_base_url}?domain=${service_config['domain']}&service=${service_config['service']}\"", returnStdout: true)
+		if(environment_data) {
+			def environment_dataObj = parseJson(environment_data)
+			if(environment_dataObj && environment_dataObj.data && environment_dataObj.data.environment) {
+				def env_collection = environment_dataObj.data.environment
+				if(env_collection.size() > 0){
+					for(_env in env_collection){
+						if( _env.logical_id == logical_id ){
+							branchName = _env.physical_id
+						}
+					}
+				}
+			}
+		}
+        return branchName
+	}
+	catch(e){
+		echo "error occured while finding branch name of $logical_id: " + e.getMessage()
+		return branchName
+	}
+}
+ 
+def generateEnvironmentMap(status, environment_logical_id) {
+
+	def env_logical_id
+	if(environment_logical_id == null) {
+		env_logical_id = getEnvironmentLogicalId()
+	} else {
+		env_logical_id = environment_logical_id
+	}
+
+	def serviceCtxMap = [
+		status: status,
+		domain: service_config['domain'],
+		branch: g_service_branch,
+		logical_id: env_logical_id		
+	]
+	if(g_environment_endpoint != null) {
+		serviceCtxMap.endpoint = g_environment_endpoint
+	}
+
+	return serviceCtxMap;
+}
+
+def generateDeploymentMap(status, environment_logical_id) {
+
+	def env_logical_id
+	if(environment_logical_id == null) {
+		env_logical_id = getEnvironmentLogicalId()
+	} else {
+		env_logical_id = environment_logical_id
+	}
+
+	def serviceCtxMap = [
+		environment_logical_id: env_logical_id,
+		status: status,
+		domain: service_config['domain'],
+    	provider_build_url:  g_build_url,
+    	provider_build_id: g_build_id,
+    	scm_commit_hash:  getRepoCommitHash(),
+    	scm_url:  "${getRepoURL()}.git",
+    	scm_branch: g_service_branch,
+		request_id: g_request_id
+	]
+	return serviceCtxMap;
+}
+
+def generateDeleteDeploymentMap() {
+
+	def serviceCtxMap = [
+		domain: service_config['domain'],
+    	provider_build_url:  g_build_url,
+    	provider_build_id: g_build_id
+ 	]
+	return serviceCtxMap;
+}
+
+def setCredentials(catalog_sv_account) {
+	
+	withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: catalog_sv_account, passwordVariable: 'Password', usernameVariable: 'Username']]) {
+		setCredentials(Username, Password)
+	}
+}
+
+def setGitCredentials(repo_credential_id) {
+	
+	withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: repo_credential_id, passwordVariable: 'PWD', usernameVariable: 'UNAME']]) {
+		setGitCredentials(UNAME, PWD)
+	}
+}
+
+ /**
+ * For setting token to access Bitbucket APIs.
+ */
+ def getGitToken() {
```

Comment:
Is this function being used?

Reply:
Not sure since I am handling only the API.